### PR TITLE
Api request update

### PR DESCRIPTION
**Pull request by doluk**
Including the stuff of #123 there are some further improvements.
I added for our new Generators an optional batching. So unless you turn it off, coc.py doesn't request the full warlog/raidlog. It only does it on demand. So in case your iterating out of the current batch, it will automatically request the new batch. Works for iteration and indexing. You can turn it off by setting pagination = False. The batch size is 5 as default, but can be modified by adding limit to the kwargs.

Further the api endpoint for the current gold pass season was added as api route, and function of the client. The regarding event and miscmodel are also included.

Additional the httpClient requests has been updated to support kwargs on every endpoint which support any filter(s)

### CODE REVIEW

File: coc/client.py [hunk 1]
```diff
@@ -538,12 +541,61 @@ async def get_warlog(
...
+
+    async def get_raidlog(
+        self,
+        clan_tag: str,
+        cls: Type[RaidLogEntry] = RaidLogEntry,
+        paginated: bool = True,
+        **kwargs
+    ) -> RaidLog:
+        """Retrieve a clan's raid log.
+        Set paginated = False to get the full raid log with one API call.
```
Comment (r-priyam) [inline coc/client.py:561]:
```suggestion
        """
        Retrieve a clan's raid log.
        Set paginated = False to get the full raid log with one API call.
```

File: coc/client.py [hunk 2]
```diff
@@ -538,12 +541,61 @@ async def get_warlog(
...
+
+        Maintenance
+            The API is currently in maintenance.
+
+        GatewayError
+            The API hit an unexpected gateway exception.
+
+
+        Returns
+        --------
```
Comment (r-priyam) [inline coc/client.py:584]:
```suggestion

        Returns
        --------
```
Lines should have consistent spacing

File: coc/raid.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+
+    def _from_data(self, data):
+        data_get = data.get
+
+        self.name: str = data_get("name")
+        self.tag: str = data_get("tag")
+        self.attack_count: int = data_get("attacks")
+        self.attack_limit: int = data_get("attackLimit")
+        self.bonus_attack_limit: int = data_get("bonusAttackLimit")
+        self.capital_resources_looted: int = data_get("capitalResourcesLooted")
```
Comment (r-priyam) [inline coc/raid.py:96]: `.get` by default returns `None` if not found, so typing needs some care

File: coc/client.py [hunk 4]
```diff
@@ -1520,6 +1572,25 @@ async def verify_player_token(self, player_tag: str, token: str) -> bool:
...
+        ------
+        Maintenance
+            The API is currently in maintenance.
+
+        GatewayError
+            The API hit an unexpected gateway exception.
+
+
+        Returns
+        --------
```
Comment (r-priyam) [inline coc/client.py:1588]:
```suggestion

        Returns
        --------
```
This needs consistent docstring too

**Review by r-priyam — CHANGES_REQUESTED** [review]
I just requested some minor nitpicking

File: coc/raid.py (hunk 3 above)
Comment (doluk) [reply coc/raid.py:96]: as this is the way coc.py does it everywhere else, i would stick with it

File: coc/client.py (hunk 1 above)
Comment (doluk) [reply coc/client.py:561]: This one is also inline with the other client functions and docstrings

File: coc/client.py [hunk 5]
```diff
@@ -538,12 +541,60 @@ async def get_warlog(
...
+        self,
+        clan_tag: str,
+        cls: Type[RaidLogEntry] = RaidLogEntry,
+        paginated: bool = True,
+        **kwargs
+    ) -> RaidLog:
+        """Retrieve a clan's raid log.
+        Set paginated = False to get the full raid log with one API call.
+
+        Parameters
```
Comment (majordoobie) [inline coc/client.py:563]: Parameters are missing from the documentation. I can fix this when I merge it.

**Pull request merged**